src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -11,55 +11,64 @@
 using Bicep.Core.Resources;
 using Bicep.Core.SemanticModel;
 using Bicep.Core.Syntax;
+using Bicep.Core.Syntax.Visitors;
 
 namespace Bicep.Core.TypeSystem
 {
     public sealed class TypeAssignmentVisitor : SyntaxVisitor
     {
+        public class TypeAssignment
```

Comment:
Since it's public, can you put this class in a separate file?

Reply:
Also, we should probably have comments on the class to describe exactly what it represents.

Reply:
I've updated this to make it private. I started that way, made it public for some reason, and now appear to have no good reason to keep it public...